Topics/Tech_Stacks/Writing_Clean_React_Code.md

```diff
@@ -0,0 +1,153 @@
+
+- **Try to use fewer States**: State management is a powerful tool in React, but it can also be overused. It is important to consider whether a component really needs to be stateful. If data to persist across re-renders, but don't want every update to **cause** a re-render, you might be interested in using useRef hooks instead. Moreover, it is easier to reuse stateless components.
+
+One way to reduce the number of states is to group related states into a single state object. For example, a component that fetches user data from a database can store the user's name, email, and bio in a single state object instead of storing each piece of data in a separate state.
+
+```jsx
+// This is a bad example
+const [name, setName] = useState("");
+const [email, setEmail] = useState("");
+const [bio, setBio] = useState("");
+```
+
+```jsx
+// This is a better example
+const [user, setUser] = useState({
+  name: "",
+  email: "",
+  bio: "",
+});
+
+// It is also good to destruct the state object when using it
+const { name, email, bio } = user;
+```
+
+It is important to note that this is not always the best approach, since it can lead to a bloated state object, and its also important for everyone working in the codebase to be aware of the individual states that are grouped together in the state object.
+
+State management is a very complex topic that warrants its own article, so to learn more you might want to check out the [React documentation on state and lifecycle](https://react.dev/learn/state-a-components-memory). [This resource](https://oskari.io/blog/stop-react-global-state-part-1) might also be useful to understand the different types of states a component may have.
+
+- **Only use Effects when necessary**: Effects are useful for performing side effects such as fetching data from a database or subscribing to an event listener, however they increase the complexity of the code and thus must only be used when necessary. If a component only needs to render data, then it should not use an effect. This helps reduce the complexity of the component and makes it easier to debug. Moreover, it is easier to reuse components that do not use effects since they are not tied to a specific effect. Look at the following example taken from the amazing open source course over at [Odin Project]().
```

Comment:
I believe there should be a link to the Odin Project in the parentheses here.

Reply:
Oh, I must have missed it. Added it!